rfcs/detecting_sensitive_textual_content/20230308-implementation_plan_filtering_and_designating_results_with_sensitive_textual_content.md

```diff
@@ -0,0 +1,581 @@
+1. We explicitly refresh the index after it is finished reindexing to prime it
+   to be searched. ES only automatically refreshes indexes that are being
+   actively used, and the index should be refreshed before it starts being used.
+
+The creation of the new index follows the same pattern of creating the regular
+index: create the index, copy the data into the index, then issue the command to
+point the alias to the new index. This process is encapsulated in two actions:
+the `create_and_populate_filtered_index` action described above, and the
+existing `point_alias` action.
+
+```{note}
+While both processes have a "copy" step, they are notably different. The
+origin index receives its documents from Postgres via ingestion workers, which
+push the documents into Elasticsearch. The filtered index receives its
+documents via the reindex API and does not require the ingestion workers.
+```
+
+### Airflow data refresh DAG factory (overview step 2)
+
+To enable running filtered index creation independently of the data refresh
+DAGs, we will create a separate DAG that will be triggered by the data refresh
+DAGs using the
+[`TriggeredDagRunOperator`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/trigger_dagrun/index.html).
+The new DAG will be named `<media_type>_filtered_index_creation`[^passive-name].
+It will not run on a schedule (only triggered), it will have a max concurrency
+of one, and it will accept the following configuration variables:
+
+[^passive-name]:
+    The passive voice name of this DAG is not ideal, but
+    `create_filtered_<model_name>_index_dag` reads as a DAG factory function
+    rather than a dag named "create filtered index".
+
+- `model_name`: The model being filtered. Used as the alias of the origin index
+  to use.
+- `index_suffix`: Optionally, the suffix to add to the destination index.
+- `force`: Used by the data refresh DAG to override the locking behaviour of the
+  DAG that prevents it from running during a data refresh.
+
+Splitting the process into a separate DAG presents complexities that are
+reflected in the last variable, `force`. Consider the following situation: a
+data refresh is currently happening, but you want to create a new filtered
+index. You can theoretically kick off the reindex job from the existing index.
+However, after the data refresh is finished, the previous index will be deleted!
+This will prevent the reindex from completing successfully. Therefore, we cannot
+run filtered index creation if a data refresh is currently underway.
+Additionally, if filtered index creation is underway, then we cannot run a data
+refresh either. Even though the data refresh process takes considerable time
+before it's ready to delete the previous index, we don't want to play with race
+conditions.
+
+Rather than trying to use a lock, we will read the DAG status directly from the
+DAG and fail the filtered index creation DAG run if the corresponding data
```

Comment:
Can you clarify which DAG `the DAG` is here? I think it's the data refresh DAG but clarity would be helpful:
```suggestion
Rather than trying to use a lock, we will read the DAG status directly from the
data refresh DAG and fail the filtered index creation DAG run if the corresponding data
```

Reply:
I think the wording is confusing. I'll change it to "we will read the data refresh DAG status at the start of the create filtered index DAG" to remove the ambiguity.